helix-term/src/ui/editor.rs

```diff
@@ -532,7 +532,11 @@ impl EditorView {
             }
             _ => {
                 // set the count
-                cxt.count = cxt.editor.count.take().unwrap_or(1);
+                if let Some(count) = cxt.editor.count.take() {
+                    if let Some(count) = std::num::NonZeroUsize::new(count) {
+                        cxt.count = Some(count);
+                    }
+                }
```

Comment:
I suggest to also change Editor's `count` to be a `NonZeroUsize` and simply do

```suggestion
                cxt.count = cxt.editor.count.take()
```